Sources/Stencil/Loader.swift

```diff
@@ -51,27 +52,19 @@ public class FileSystemLoader: Loader, CustomStringConvertible {
   }
 
   public func loadTemplate(name: String, environment: Environment) throws -> Template {
-    for path in paths {
-      let templatePath = try path.safeJoin(path: Path(name))
-
-      if !templatePath.exists {
-        continue
-      }
-
-      let content: String = try templatePath.read()
-      return environment.templateClass.init(templateString: content, environment: environment, name: name)
-    }
-
-    throw TemplateDoesNotExist(templateNames: [name], loader: self)
+    return try loadTemplate(names: [name], environment: environment)
   }
 
   public func loadTemplate(names: [String], environment: Environment) throws -> Template {
     for path in paths {
       for templateName in names {
-        let templatePath = try path.safeJoin(path: Path(templateName))
+        let templatePath = URL(fileURLWithPath: templateName, relativeTo: URL(fileURLWithPath: path))
+        if !templatePath.withUnsafeFileSystemRepresentation({ String(cString: $0!) }).hasPrefix(path) {
```

Comment:
Is this equivilent to the old implementation? There doesn't seem to be an explict asolute path, perhaps this is happening for free by this construct?

There also doesn't appear to be any test cases for `SuspiciousFileOperation`.

Reply:
Yes, it should be equivalent to the previous implementation.  The `pathJoin` is concatenating two path components.  The FSR is going to the canonicalised concatenation of the paths.  The current tests do invoke this method IIRC and do pass.

Reply:
I'd rather have this in a tiny extension on `URL` or `FileManager`, instead of inside the `loadTemplate` function.